New role cleaned

pwalczysko opened the pull request:
# What this PR does
This PR supersedes https://github.com/openmicroscopy/openmicroscopy/pull/5048.

# What this PR does

Adds a basic set of workflow-based java tests (server-side) to the new roles features.

The accompanying tests deal with two problematics:

# Testing this PR

Mainly check that the accompanying tests as well as all the other integration tests are passing and make sense.
The tests here are setting up the lightAdmin always in following manner:
   - take all the privileges away from the newly set up lightAdmin (see the [list of privileges](https://github.com/openmicroscopy/openmicroscopy/blob/roles/components/model/resources/mappings/meta.ome.xml#L180)
   - give to the lightAdmin "back" only the privileges which are being investigated in the particular test
   - using the test, see if the lightAdmin, equipped with this minimal set of privileges, is able to perform a meaningful workflow (such as ImportAs, Delete others' data, Edit others' Project etc.)

For understanding of particual tests, see the [explanatory ppt](https://docs.google.com/presentation/d/18kc75LHiItkHr_dvG1R6jW1qmTNayan8x4ZQC6s7XmI/edit#slide=id.g1b11290c7c_2_96). 
   - the link should bring you to the slide 3 of the ppt
   - on this slide, there is the list of tests
   - follow the links to the single test explanatory ppts
   - only the complex tests have links to graphical explanatory ppts, the simple tests (12. - 21.) shoiuld be readable from the code itself


# Related reading

[design issue](https://github.com/openmicroscopy/design/issues/62)
[trello card](https://trello.com/c/t0nT7KYa/133-new-role)
Can* methods are tested in #5248 
Annotations test are added in https://github.com/openmicroscopy/openmicroscopy/pull/5275

cc @jburel @joshmoore @mtbc @bramalingam

pwalczysko committed:
Add test for ImporterAs workflow, adjust parent class

pwalczysko committed:
Delete unused imports, change author, delete unused methods

pwalczysko committed:
Add tests for ImporterAs ability to link and delete data with Sudo only

pwalczysko committed:
Add tests for edit name and chgrp for importerAs Sudo only

pwalczysko committed:
Add test for additional chgrp privilege on chgrp behaviour

pwalczysko committed:
Add test for ImporterAs using Chown permission after import

pwalczysko committed:
Simplify the chown test, add similar logic to chgrp test, catch server bug

pwalczysko committed:
Make the tests work with recent Marks changes in parent clesses

pwalczysko committed:
Adjust chgrp test for ImporterAs and aligh it with chown test

pwalczysko committed:
Delete print statements, correct comments

pwalczysko committed:
Add tests for ImporterAs workflows with No Sudo

pwalczysko committed:
Add tests for creation and targeting of Dataset to NoSudo ImporterAs workflows

pwalczysko committed:
Add test for Delete behavior of a sudoing and non-sudoing light admin

pwalczysko committed:
Add a supplementary test for light admin linking users objects

pwalczysko committed:
Hieararchy chown failing for light admin, passing for full admin, also psssing if light admin owns hierarchy

pwalczysko committed:
Permutate over all group types in combined privilege tests

pwalczysko committed:
Adjust tests to test Mark's fix of the chown hierarchy bugs

pwalczysko committed:
Fix the testImporterAsNoSudoLinkInTargetGroup : the full hierarchy chown does not work in this case

pwalczysko committed:
Rename test accordingly to its functions

pwalczysko committed:
Fix the comment in the test according to test function

pwalczysko committed:
Widen Edit workflow test for NoSudo, permutate over all groups and permission combinations

pwalczysko committed:
Fix the lightAdminEdit test for ownership of the Project

pwalczysko committed:
Go over all group permissions in the testImporterAsSudoCreateImport

pwalczysko committed:
Adjust the testDataOrganizerChownAll to truly chown all belonging to a user

pwalczysko committed:
Remove redundant group context specifications cf. Mark's comments

pwalczysko committed:
Adjust data provider description

pwalczysko committed:
Make chown passing test more explicit in testDataOrganizerChownAll

pwalczysko committed:
Add ownership checks for original file in testImporterAsSudoChown, add explanation to testDataOrganizerChownAll

pwalczysko committed:
Improve explanatory comments in testImporterDelete

pwalczysko committed:
Make the testImporterAsSudoChgrp more concise, improve comments

pwalczysko committed:
Check also for original file group in testImporterAsSudoChgrp

pwalczysko committed:
Add tests for no sudo workflow to Create part of testImporterAsSudoCreateImport

pwalczysko committed:
Fix testDataOrganizerChownAll test to pass after Marks additions to permission granularity

pwalczysko committed:
Temporary commit to delete or squash later

pwalczysko committed:
Fix testImporterAsNoSudoChownOnly and testImporterAsSudoChown for new Marks features

pwalczysko committed:
Fix the comments in two tests

pwalczysko committed:
Add check for OrigFile to testImporterAsNoSudoChownOnlyWorkflow and fix testImporterAsSudoChown for new Marks permissions

pwalczysko committed:
Fix testImporterDelete for new permissions added by Mark

pwalczysko committed:
Adjust the testImporterAsSudoChown for the new behaviour of Chown (draws with it all the other necessary permissions)

pwalczysko committed:
Fix testImporterAsNoSudoChownOnlyWorkflow for new permissions added by mark with changed chown width

pwalczysko committed:
Improve comments on testImporterAsSudoChown

pwalczysko committed:
Fix testDataOrganizerChownAll for new chown behaviour and adjust comments

pwalczysko committed:
Fix testImporterDelete for new permissions, simplify

pwalczysko committed:
Fix testImporterAsNoSudoChgrpChownWorkflow for new permissions

pwalczysko committed:
Adjust testImporterAsSudoChgrp for new permissions, simplify

pwalczysko committed:
Simplify testLightAdminEdit for new permissions

pwalczysko committed:
Simplify testImporterAsSudoCreateImport for new permissions setup

pwalczysko committed:
Simplify testImporterAsNoSudoLinkInTargetGroup for new permissions setup

pwalczysko committed:
Remove unused data providers

pwalczysko committed:
Remove extra permission given in testImporterAsNoSudoChgrpChownWorkflow

pwalczysko committed:
Rename all the tests to match the content, improve one comment

pwalczysko committed:
Fix testDelete cf. Mark's comments

pwalczysko committed:
Add "severing ot the link" workflow to testChgrp

pwalczysko committed:
Add "severing of the link" workflow to testChown

pwalczysko committed:
Improve testLinkNoSudo description

pwalczysko committed:
Add test for rendering settings creation and chown by light admin

pwalczysko committed:
Improve logic of testRenderingSettingsNoSudo

pwalczysko committed:
Fix typo

pwalczysko committed:
Add test for ROI creation and chowning

pwalczysko committed:
Replace the declaration in all tests cf. Mark's comment

pwalczysko committed:
Fix double semicolon typo

pwalczysko committed:
Swap the order of image creation and light admin establishment cf. Mark's comment

pwalczysko committed:
Remove the group context line as the user is already in the group

pwalczysko committed:
Add test for File Attachment creation and linking

pwalczysko committed:
Rename the testFileAttachmentNoSudo (no tags are being handled in this test)

pwalczysko committed:
Add test for upload of official script

pwalczysko committed:
Add test for Deleting of official script

pwalczysko committed:
Add test for adding user into group via iAdmin.addGroups

pwalczysko committed:
Add test for removal of a user from a group

pwalczysko committed:
Add test for setting a new group owner

pwalczysko committed:
Add test for unsetting Group Owner

pwalczysko committed:
Add test for creation of new user

pwalczysko committed:
Add test for editing a user

pwalczysko committed:
Add test for group creation

pwalczysko committed:
Correct the privilege line in three tests

pwalczysko committed:
Add test for group editing

pwalczysko committed:
Improve javadoc, spacing and remove unused imports

pwalczysko committed:
Add more precise provider for testCreateLinkImport and testEdit

pwalczysko committed:
Add more precise provider for testDelete, improve comments in the test

pwalczysko committed:
Add more precise provider for testChgrp, improve comments in the test

pwalczysko committed:
Add more precise provider for testChown, improve comments in the test

pwalczysko committed:
Add more precise provider for testImporterAsNoSudoChownOnlyWorkflow, improve comments in the test

pwalczysko committed:
Add more precise provider for testLinkNoSudo, improve test comments and structure

pwalczysko committed:
Add more precise provider for testROIAndRenderingSettingsNoSudo, improve comments in the test

pwalczysko committed:
Add more precise provider for testImporterAsNoSudoChgrpChownWorkflow

pwalczysko committed:
Add more precise provider to testChownAllBelongingToUser, remove redundant parameter from the test

pwalczysko committed:
Add more precise provider to all the rest of the tests

pwalczysko committed:
Rename the variables in simpler tests to isPrivileged to match the logic in Mark's tests

pwalczysko committed:
Remove unused data providers, improve comments on the other providers

pwalczysko committed:
Remove isAdmin from the tests

pwalczysko committed:
Add assertOwnedBy to all test

pwalczysko committed:
Edit names of tests, adjust comments

pwalczysko committed:
Correct comment

joshmoore commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+/*
+ * Copyright (C) 2016 University of Dundee & Open Microscopy Environment.
```
2017

joshmoore commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+/**
+ * Tests the concrete workflows of the light admins
+ * @author p.walczysko@dundee.ac.uk
+ * @since 5.3.0
```
5.4.0

joshmoore commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        if (isSudoing) {
+            try {
+                sudo(new ExperimenterI(normalUser.userId, false));
+                }catch (SecurityViolation sv) {
```
minor whitespace typo

joshmoore reviewed (commented):
In general this is _very_ impressive. Long-term I'm slightly worried about the maintainability. A few general suggestions which aren't necessarily intended for immediate action:

 * A few of the methods (e.g. `linkProjectDataset`) either are or could be at a higher level in the class hierarchy.
 * For a new reader of this code, the details and tested steps in the `test*` methods are fairly interwoven.
 * Any refactoring for the above might also taken into account trying to bring the PPT and/or spreadsheet representations of what's being tested closer in line with this one.

mtbc commented on components/tools/OmeroJava/test/integration/AbstractServerImportTest.java:
```diff
@@ -70,7 +71,7 @@ protected ImportLocation importFileset(List<String> srcPaths) throws Exception {
      *             unexpected
      */
-    protected ImportLocation importFileset(List<String> srcPaths, int numberToUpload) throws Exception {
+    protected ImportLocation importFileset(List<String> srcPaths, int numberToUpload, IObject targetObject) throws Exception {
```
add a `@param` for this -- there's a setting in Eclipse that will warn of such missing stuff

pwalczysko replied:
Fixed in https://github.com/openmicroscopy/openmicroscopy/pull/5159/commits/03c450801095d4b7cbdef305454f58aeb5d45114

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+     * All workflows are tested here both when light admin is sudoing
+     * and when he/she is not sudoing, except for Link and Import (both tested
+     * only when sudoing, as the non-sudoing workflows are too complicated
+     * for those two actions and thus covered by separate tests.
```
close parenthesis

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        final EventContext lightAdmin;
+        lightAdmin = loginNewAdmin(true, permissions);
+        if (isSudoing) {
+            try {
```
do we need this `try` at all?

pwalczysko replied:
Will fix at a later stage, captured in first checklist point of https://trello.com/c/ygDN23CC/41-new-role-integration-tests-petr

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        client.getImplicitContext().put("omero.group", Long.toString(normalUser.groupId));
+        Project proj = mmFactory.simpleProject();
+        Dataset dat = mmFactory.simpleDataset();
+        Project sentProj = new ProjectI();
```
we just throw away this new object

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+     * user. Behaviors of the system are explored when light admin
+     * is and is not using <tt>Sudo</tt> privilege
+     * for this action.
+     * @param isAdmin if to test a member of the <tt>system</tt> group
```
Javadoc again not matching.

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+       if (permDeleteOwned) permissions.add(AdminPrivilegeDeleteOwned.value);
+       final EventContext lightAdmin;
+       lightAdmin = loginNewAdmin(true, permissions);
+       try {
```
will always succeed?

pwalczysko replied:
Will fix at a later stage, captured in first checklist point of https://trello.com/c/ygDN23CC/41-new-role-integration-tests-petr

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+       client.getImplicitContext().put("omero.group", Long.toString(normalUser.groupId));
+       Project proj = mmFactory.simpleProject();
+       Dataset dat = mmFactory.simpleDataset();
+       Project sentProj = new ProjectI();
```
created but thrown away

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+           List<String> path = Collections.singletonList(fakeImageFile.getPath());
+           importFileset(path, path.size(), sentDat);
+       } catch (ServerError se) {
+       }
```
might this error happen then? do we want to assert anything about when?

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        * would delete the whole hierarchy, which was successfully tested
+        * during writing of this test.*/
+       if (deletePassing) {
+           doChange(Requests.delete().target(datasetImageLink).build());
```
use `target(IObject... targets)` for all at once? or if that doesn't work, can just do `target(foo).target(bar)...`

mtbc replied:
After conversation: ah, maybe just extend above comment to make it clearer that you positively want to try piecemeal deletion.

mtbc commented:
A few of my comments look to apply similarly to later tests.

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+     * to sever necessary links for performing the chgrp. This is achieved by
+     * having the image which is getting moved into a different group in a dataset
+     * in the original group (the chgrp has to sever the DatasetImageLink to perform
+     * the move (chgrp).
```
close outer parenthesis

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        remoteFileGroupId = ((RLong) iQuery.projection(
+                "SELECT details.group.id FROM OriginalFile o WHERE o.id = :id",
+                new ParametersI().addId(remoteFileId)).get(0).get(0)).getValue();
+        if(chgrpNoSudoExpectSuccessAnyGroup) {
```
`if (...`

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        if(chgrpNoSudoExpectSuccessAnyGroup) {
+            /* check that the image moved to another group */
+            Assert.assertEquals(afterSecondChgrpImageGroupId, anotherGroupId);
+            Assert.assertEquals(afterSecondChgrpImageGroupId, anotherGroupId);
```
these look remarkably similar (also below), was one meant to be owner?

pwalczysko replied:
This is fixed in https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R792 and https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R914 - the testChgp was split in the latter PR into two separate tests, the check for the annotations was added as well as a check for the OriginalFile.

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+                    "SELECT details.group.id FROM Image i WHERE i.id = :id",
+                    new ParametersI().addId(image.getId())).get(0).get(0)).getValue();
+            Assert.assertEquals(imageGroupId, normalUser.groupId);
+            assertOwnedBy(image, normalUser);
```
I wonder if this final stuff here can be made enough similar to the `else` case to be done outside the `if`

pwalczysko replied:
I think I addressed this in https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R790 and https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R910 by splitting the testChgrp in two tests and having only one if + else per each of those tests. The content of if and else blocks is very dissimilar on this latter branch

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+     * is a member of, then just chowned to the user.
+     * This workflow is possible only if PR#4957 dealing with
+     * admins importing data into groups they are not member of will get
+     * merged. For this test, combinations of  <tt>Chown</tt>, <tt>WriteOwned</tt>,
```
double-space! :scream: (more below, just search)

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        if (importNotYourGroupExpectSuccess) {
+            assertOwnedBy(remoteFile, lightAdmin);
+            Assert.assertEquals(remoteFile.getDetails().getGroup().getId().getValue(), normalUser.groupId);
+        } else Assert.assertNull(remoteFile, "if import failed, the remoteFile should be null");
```
maybe put in its own `{...}` or at least on next line

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        } else Assert.assertNull(remoteFile, "if import failed, the remoteFile should be null");
+        /* check that also the image corresponding to the original file is in the right group */
+        Image image = null;
+        if (!(remoteFile == null)) {
```
more idiomatically,
```java
if (remoteFile != null) {
```

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        /* linking should be always permitted as long as light admin is in System Group
+         * and has WriteOwned permissions. Exception is Private group, where linking will
+         * always fail.*/
+        boolean isExpectLinkingSuccess = permWriteOwned && !(groupPermissions == "rw----");
```
`!"rw----".equals(groupPermissions)` is safest

(also below)

pwalczysko replied:
https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R959
https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R1676
https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R1954

are the fixes for this, which are on a latter PR.

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        Chown2 chown = Requests.chown().target(linkOfDatasetImage).toUser(normalUser.userId).build();
+        doChange(client, factory, chown, isExpectSuccessLinkAndChown);
+        chown = Requests.chown().target(linkOfProjectDataset).toUser(normalUser.userId).build();
+        doChange(client, factory, chown, isExpectSuccessLinkAndChown);
```
FYI can target both links in same chown if desired

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        assertOwnedBy(sentDat, normalUser);
+        assertOwnedBy(sentProj, normalUser);
+        if (isExpectSuccessLinkAndChown) {
+            assertOwnedBy((new DatasetImageLinkI (linkDatasetImageId, false)), normalUser);
```
no space in `DatasetImageLinkI (linkDatasetImageId`

(also more below)

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        Dataset dat = mmFactory.simpleDataset();
+        Dataset sentDat = new DatasetI();
+        sentDat = null;
+        sentDat = (Dataset) iUpdate.saveAndReturnObject(dat);
```
```java
Dataset sentDat = (Dataset) iUpdate.saveAndReturnObject(dat);
```
suffices, can remove previous two lines

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+         */
+        if (permChgrp) {
+            doChange(client, factory, Requests.chgrp().target(sentDat).toGroup(normalUser.groupId).build(), true);
+
```
remove blank line

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+            doChange(client, factory, Requests.chgrp().target(sentDat).toGroup(normalUser.groupId).build(), true);
+
+        } else {
+            doChange(client, factory, Requests.chgrp().target(sentDat).toGroup(normalUser.groupId).build(), false);
```
if you wish, you can replace this whole if...else with,
```java
doChange(client, factory, Requests.chgrp().target(sentDat).toGroup(normalUser.groupId).build(), permChgrp);
```

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+         * A successful chowning of the dataset will chown the linked image
+         * and the link too.*/
+        if (importYourGroupAndChgrpAndChownExpectSuccess) {/* whole workflow2 succeeded */
+            doChange(client, factory, Requests.chown().target(sentDat).toUser(normalUser.userId).build(), true);
```
this is simply,
```java
doChange(Requests.chown().target(sentDat).toUser(normalUser.userId).build());
```

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+            datasetImageLinkGroupId = ((RLong) iQuery.projection(
+                    "SELECT details.group.id FROM DatasetImageLink WHERE parent.id = :id",
+                    new ParametersI().addId(sentDat.getId())).get(0).get(0)).getValue();
+            /* image, dataset and link are in the normalUser's group and belong to normalUser */
```
can the above stuff precede the `if` branching? (replacing the doChange true with some suitable expectsuccess thing)

pwalczysko replied:
Yes, planning to do something similar as in https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R907. 
See second point on checklist https://trello.com/c/ygDN23CC/41-new-role-integration-tests-petr

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+            Assert.assertEquals(datasetGroupId, normalUser.groupId);
+            assertOwnedBy((new DatasetImageLinkI (datasetImageLinkId, false)), normalUser);
+            Assert.assertEquals(datasetImageLinkGroupId, normalUser.groupId);
+        } els
```
trying to decide if lowercase `other` is convention-violating or easier to read in these names, any opinion @joshmoore? (looks a bit like "image10..." in some fonts)

pwalczysko replied:
Not getting this one, sorry. Where is the lowercase ``other`` ? What does look like Image10 ?

mtbc replied:
Weird, not sure how the comment landed there! Looks like something went wrong at GH's end given the quoted `els` (sic). Try searching for `image1Other`.

pwalczysko replied:
Fixed in https://github.com/openmicroscopy/openmicroscopy/pull/5159/commits/96eb7fd0dedd974493f23a1856eb162795e6d7f4

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+            Assert.assertEquals(datasetGroupId, normalUser.groupId);
+            assertOwnedBy((new DatasetImageLinkI (datasetImageLinkId, false)), normalUser);
+            Assert.assertEquals(datasetImageLinkGroupId, normalUser.groupId);
+        } els
```
insufficient (also below)

pwalczysko replied:
fixed in https://github.com/openmicroscopy/openmicroscopy/pull/5159/commits/3917d9c251f02fdee0f88fd09eb6a6e08928ac81

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+            Assert.assertEquals(datasetGroupId, normalUser.groupId);
+            assertOwnedBy((new DatasetImageLinkI (datasetImageLinkId, false)), normalUser);
+            Assert.assertEquals(datasetImageLinkGroupId, normalUser.groupId);
+        } els
```
I hope that #5284 makes this skip unnecessary. :smiley:

pwalczysko replied:
See third point in the checklist of https://trello.com/c/ygDN23CC/41-new-role-integration-tests-petr

mtbc commented:
Given that you have other branches that build on these commits, it is fine with me if fixes follow in later PRs rather than necessarily blocking this.

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+            Assert.assertEquals(datasetGroupId, normalUser.groupId);
+            assertOwnedBy((new DatasetImageLinkI (datasetImageLinkId, false)), normalUser);
+            Assert.assertEquals(datasetImageLinkGroupId, normalUser.groupId);
+        } els
```
For this kind of simple query one may use something like,
```java
iQuery.get("FileAnnotation", fileAnnotation.getId())
```

pwalczysko replied:
Have to admit that I cannot find the place in the diff to which this comment is relevant (confused by not seeing any "FileAnnotation" around this comment). But I guess it is about queries retrieving *GroupId. These will be replaced by the assertInGroup method https://github.com/openmicroscopy/openmicroscopy/pull/5275/files#diff-d91232ef1e245a87e311c4a79a2a5f77R195 in the later PR

pwalczysko replied:
Aha, sorry, I think you meant https://github.com/openmicroscopy/openmicroscopy/pull/5159/commits/76b27570425613fc9452148e28071d183f87b366#diff-d91232ef1e245a87e311c4a79a2a5f77L1491
I realized it is not necessary, as the assertOwnedBy does the retrieval itself. Removed it in https://github.com/openmicroscopy/openmicroscopy/pull/5159/commits/76b27570425613fc9452148e28071d183f87b366

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+            Assert.assertEquals(datasetGroupId, normalUser.groupId);
+            assertOwnedBy((new DatasetImageLinkI (datasetImageLinkId, false)), normalUser);
+            Assert.assertEquals(datasetImageLinkGroupId, normalUser.groupId);
+        } els
```
maybe start such comments on a new line and use `*` along their left margin

pwalczysko replied:
Does this fall into the "interwovenness" remarked by @joshmoore ?  https://github.com/openmicroscopy/openmicroscopy/pull/5159#pullrequestreview-36972406
Will ask you tomorrow about the whole tactics here if I can.

mtbc replied:
My guess is "yes" except again (in two different browsers) my comment is now being shown for me with nothing beyond an `els`, bah.

pwalczysko replied:
I will deal with this as part of https://trello.com/c/ygDN23CC/41-new-role-integration-tests-petr, checklist "Josh's comments", point 2

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+            Assert.assertEquals(datasetGroupId, normalUser.groupId);
+            assertOwnedBy((new DatasetImageLinkI (datasetImageLinkId, false)), normalUser);
+            Assert.assertEquals(datasetImageLinkGroupId, normalUser.groupId);
+        } els
```
nice to have these DEBUG convenience helpers matching their actual providers

pwalczysko replied:
Fixed in https://github.com/openmicroscopy/openmicroscopy/pull/5159/commits/4ca0897a2ca0b9e464e5972c599dc040a92579ba

mtbc commented on components/tools/OmeroJava/test/integration/LightAdminRolesTest.java:
```diff
@@ -0,0 +1,2222 @@
+        }
+        /*in order to find the image in whatever group, get context with group
+         * set to -1 (=all groups) */
+        client.getImplicitContext().put("omero.group", Long.toString(-1));
```
this is simply,
```java
client.getImplicitContext().put("omero.group", "-1");
```

pwalczysko replied:
fixed in https://github.com/openmicroscopy/openmicroscopy/pull/5159/commits/2cbddc73fc35666b53d3d626813a38fac1ec688f

pwalczysko committed:
Fix typos cf. Josh's comments

pwalczysko committed:
Fix Mark's comment in AbstractServerImportTest.java file

pwalczysko commented:
@joshmoore : your (more high-level)  comments are captured in the first checklist https://trello.com/c/ygDN23CC/41-new-role-integration-tests-petr

pwalczysko committed:
Fix first batch of Mark's comments

pwalczysko committed:
Remove redundant try-catch from testDelete

pwalczysko committed:
Fix second batch of Mark's comments

pwalczysko committed:
Fix all the remaining double-spaces in the code

pwalczysko committed:
Fix the third batch of Mark's comments

pwalczysko committed:
Do not use space after e.g. ImageI() - fix for the whole file

pwalczysko committed:
Fix redundant @param isAdmin in javadoc in all the tests

pwalczysko committed:
Remove 2 redundant sentDataset variable declarations

pwalczysko committed:
Remove a redundant if..else

pwalczysko committed:
unsufficient -> insufficient

pwalczysko committed:
long.toString(-1) -> "-1"

pwalczysko committed:
Remove redundant query

pwalczysko committed:
Fix DEBUG statements in providers

pwalczysko commented:
I think I addressed all the comments except https://github.com/openmicroscopy/openmicroscopy/pull/5159#discussion_r115516356 which I do not get (probably because of the line shift after I have commited my changes) and https://github.com/openmicroscopy/openmicroscopy/pull/5159#discussion_r115711825 which seem to me like a wider topic, need to ask @mtbc

pwalczysko committed:
Refactor image1Other... patterned variables to avoid ambiguation with ...10...

pwalczysko commented:
This is now dealt with - all the comments were either fixed or fixed in latter PRs or are on https://trello.com/c/ygDN23CC/41-new-role-integration-tests-petr checklists.

mtbc commented:
Running time is now down to under 13min. :+1:
